File: playground2/app/user/controller/user.controller.js
const userService = require('../service/user.service');

async function registerUser(req, res, next) {
    const{email, password} = req.body;
    try{
        const user = await userService.registerUser(email, password);
        return res.status(201).json({email: user.email});
    } catch (err) {
        next(err);
    }
}

async function login(req, res, next) {
    const{email, password} = req.body;
    try{
        const tokens = await userService.login(email, password);
        return res.status(200).json(tokens);
    } catch (err) {
        next(err);
    }
}

async function generateAccessToken(req, res, next) {
    const token = req.body.token;
    try {
        const newAccessToken = await userService.generateAccessToken(token);
        return res.status(200).json({newAccessToken});
    } catch (err) {
        next(err);
    }
}

async function getProfile(req, res, next) {
    const token = req.headers['authorization']?.split(' ')[1];
    try{
        const user = await userService.getProfile(token);
        return res.status(200).json({email: user.email});
    } catch (err) {
        next(err);
    }
}

module.exports = {
    registerUser,
    login,
    generateAccessToken,
    getProfile
}

File: playground3/src/main/java/com/example/playground3/controller/UserController.java
package com.example.playground3.controller;

import com.example.playground3.model.User;
import com.example.playground3.service.UserService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/users")
public class UserController {

    private final UserService userService;

    public UserController(UserService userService) {
        this.userService = userService;
    }

    @PostMapping("/register")
    public ResponseEntity<Map<String, String>> registerUser(@RequestBody Map<String, String> body) {
        String email = body.get("email");
        String password = body.get("password");
        User user = userService.registerUser(email, password);
        return ResponseEntity.status(HttpStatus.CREATED).body(Map.of("email", user.getEmail()));
    }

    @PostMapping("/login")
    public ResponseEntity<Map<String, String>> login(@RequestBody Map<String, String> body) {
        String email = body.get("email");
        String password = body.get("password");
        Map<String, String> tokens = userService.login(email, password);
        return ResponseEntity.ok(tokens);
    }

    @PostMapping("/refresh")
    public ResponseEntity<Map<String, String>> generateAccessToken(@RequestBody Map<String, String> body) {
        String token = body.get("token");
        String newAccessToken = userService.generateAccessToken(token);
        return ResponseEntity.ok(Map.of("newAccessToken", newAccessToken));
    }

    @GetMapping("/me")
    public ResponseEntity<Map<String, String>> getProfile(@RequestAttribute("userEmail") String userEmail) {
        return ResponseEntity.ok(Map.of("email", userEmail));
    }
}
